### DISCUSSION TITLE
For noise reduction, ‘DeepFilterNet2’ should be implemented in ‘Deep Noise Remover’ instead of ‘DeepFilterNet’.

### DISCUSSION THREAD

**Discussion by ja1bbp** [wwmm/easyeffects#3141] [Ideas]
Currently, the functionality of 'Deep Noise Remover' is realized by 'DeepFilterNet'. However, the load of ‘DeepFilterNet’ is heavy and there are delays. Do you have any plans to implement 'DeepFilterNet2', which aims to reduce the load and improve real-time performance, in 'Deep Noise Remover'? I look forward to that happening.

**Comment by wwmm** [comment]
I was not aware of this new implementation. The library page has no release with it https://github.com/Rikorose/DeepFilterNet. If the library is updated we will do the necessary adjustments on our side.

**Reply by ja1bbp** [reply]
I really hope that "DeepFilterNet2" will be implemented instead of "Deep Noise Remover". I have confirmed that running "Deep Noise Remover" on Raspberry Pi5 increases the CPU load by 40-50%. "Deep Noise Remover" alone takes up that much CPU resource. It is because of these circumstances that I have high expectations for "DeepFilterNet2." 
Thank you for your support.

**Reply by rmeissn** [reply]
@wwmm Just a little update on the releases page of https://github.com/Rikorose/DeepFilterNet/releases
- DeepFilterNet2 was released with version 0.2.3 in May 2022
- DeepFilterNet3 was release with version 0.5.0 in June 2023
- The pypi package (I guess this is the library?) is on par with the latest release on Github

**Reply by ja1bbp** [reply]
Thanks for the info. That's good news!

I hope that DeepFilterNet2 or 3 will be implemented to achieve real-time control. The current "Deep Noise Remover(DeepFilterNet)" takes 40ms to run.

**Reply by hayama77** [reply]
Logically considering the final release date of DeepFilterNet (2023-9-1) and the date DeepFilterNet was implemented in EasyEffects "version 7.1.2" (2023-11-08), I have come to the conclusion that it is reasonable to conclude that DeepFilterNet3 is already implemented in the current EasyEffects.  (^_^

**Comment by hayama77** [comment]
Hello everyone!
I would like to inform you that sampling files comparing DeepFilterNet2 and other noise reduction software are available on the following site.
Previously, I have used RNNoise, but when I came across DeepFilterNet, I was very surprised. And now I'm even more surprised to find DeepFilterNet2. 

I hope that the information on the following sites will be helpful to you, and someone implements DeepFilterNet2 into Easy Effects.   (^_^)

https://rikorose.github.io/DeepFilterNet2-Samples/

**Reply by wwmm** [reply]
>and someone implements DeepFilterNet2 into Easy Effects

It is enough if someone implements (and maintains) a library based onit it or if the one already available for "deepfilternet" is updated to this new algorithm. Reading the article and writing a library based on it is probably out of my capabilities...

**Reply by hayama77** [reply]
> It is enough if someone implements (and maintains) a library based onit it or if the one already available for "deepfilternet" is updated to this new algorithm.

Yes, Let's wait for that time to come！

**Comment by hayama77** [comment]
Hi!
The current list of plug-ins does not list 'Deep Noise Remover'. Therefore, there is also no descriptive text about DeepFilterNet. Who is the editor of the plugin list?　By the way, I have created an explanatory document about 'Deep Noise Remover'. It may not be perfect, but I can provide it if it would be helpful to someone.   (^_^)/

**Reply by wwmm** [reply]
>The current list of plug-ins does not list 'Deep Noise Remover'. Therefore, there is also no descriptive text about DeepFilterNet.

I totally forgot to add it there.

>Who is the editor of the plugin list?

No one in particular. I or other collaborators do it when we remember. If you want to do a pull request updating our readme page with this filter information feel free to do so.

**Reply by hayama77** [reply]
This is my first attempt to post a 'pull request'. 
I don't know if it worked, but I did it anyway.
Please confirm it.

**Reply by wwmm** [reply]
> This is my first attempt to post a 'pull request'. I don't know if it worked, but I did it anyway. Please confirm it.

It worked. Now that I am looking at it it will probably be better to have it in the same format as the docs for the other plugins https://github.com/wwmm/easyeffects/tree/master/help/C. But this is something I can do tomorrow.

**Comment by geextahslex** [comment]
So which DeepFilterNet version is now used?

**Reply by wwmm** [reply]
In my Arch Linux installation I have version 0.5.6
